src/OrasProject.Oras/Pack.cs

```diff
@@ -0,0 +1,333 @@
+                MediaType = "application/vnd.oci.empty.v1+json",
+                Digest = Digest.ComputeSHA256(expectedConfigBytes),
+                Data = expectedConfigBytes,
+                Size = expectedConfigBytes.Length
+            };
+            options.Layers.Add(emptyLayer);
+        }
+
+        var annotations = EnsureAnnotationCreated(options.ManifestAnnotations, "org.opencontainers.image.created");
+
+        var manifest = new Manifest
+        {
+            SchemaVersion = 2,
+            MediaType = "application/vnd.oci.image.manifest.v1+json",
+            ArtifactType = artifactType,
+            Subject = options.Subject,
+            Config = options.Config,
+            Layers = options.Layers,
+            Annotations = annotations
+        };
+
+        return await PushManifest(pusher, manifest, manifest.MediaType, manifest.ArtifactType, manifest.Annotations, cancellationToken);
+    }
+
+    private static async Task<Descriptor> PushManifest(ITarget pusher, object manifest, string mediaType, string? artifactType, IDictionary<string, string>? annotations, CancellationToken cancellationToken = default)
+    {
+        var manifestJson = JsonSerializer.SerializeToUtf8Bytes(manifest);
+        var manifestDesc = new Descriptor {
+            MediaType = mediaType, 
+            Digest = Digest.ComputeSHA256(manifestJson),
+            Size = manifestJson.Length
+        };
+        manifestDesc.ArtifactType = artifactType;
+        manifestDesc.Annotations = annotations;
+
+        await pusher.PushAsync(manifestDesc, new MemoryStream(manifestJson), cancellationToken);
+        return manifestDesc;
+    }
+
+    private static void ValidateMediaType(string mediaType)
+    {
+        if (!_mediaTypeRegex.IsMatch(mediaType))
+        {
+            throw new InvalidMediaTypeException($"{mediaType} is an invalid media type");
+        }
+    }
+
+private static async Task<Descriptor> PushCustomEmptyConfig(ITarget pusher, string mediaType, IDictionary<string, string> annotations, CancellationToken cancellationToken = default)
+    {
+        var configBytes = JsonSerializer.SerializeToUtf8Bytes(new { });
+        var configDescriptor = new Descriptor
+        {
+            MediaType = mediaType, 
+            Digest = Digest.ComputeSHA256(configBytes),
+            Size = configBytes.Length
+        };
+        configDescriptor.Annotations = annotations;
+
+        await PushIfNotExist(pusher, configDescriptor, configBytes, cancellationToken);
+        return configDescriptor;
+    }
+
+    private static async Task PushIfNotExist(ITarget pusher, Descriptor descriptor, byte[] data, CancellationToken cancellationToken = default)
+    {
+        await pusher.PushAsync(descriptor, new MemoryStream(data), cancellationToken);
+    }
+
+    private static IDictionary<string, string>? EnsureAnnotationCreated(IDictionary<string, string> annotations, string key)
+    {
+        if (annotations is null)
```

Comment:
nit: I'm not which one is better (`is` or `==`).
```suggestion
        if (annotations == null)
```

Reply:
Updated